### ISSUE TITLE
1.12.65: make the cluster object profile optional?

### ISSUE THREAD

**Issue by jouvin** [quattor/aquilon#124]
Currently, when  a cluster is defined, a plenary object profile is created and compiled when `aq make cluster` is executed. I have the feeling that in general this object profile is not needed as it is not consumed by standard components. On the other hand, getting it to compile successfully can be painful, in particular if features are added to the archetype or the personality. It is the case if the feature templates need variables related to the OS (like `OS_VERSION_PARAMS`) for example, as the OS configuration is not part of the cluster object.

My proposal would be to have an option to make the creation of this object profile optional. Any objection?

**Comment by jouvin** [comment]
After discussion, the problem happens probably only if you attach a host feature to a cluster archetype rather than a cluster personality as the cluster personality is not included into the cluster. Will check.

**Comment by jouvin** [comment]
Unfortunately, I checked after our discussion and the cluster personality is included in the cluster object template. For example:

```
object template clusters/openstack_devel;

variable LOADPATH = list(
  "openstack"
);

"/metadata/template/branch/name" = "svom";
"/metadata/template/branch/type" = "sandbox";
"/metadata/template/branch/author" = "jouvin";

include "archetype/declarations";

include "pan/units";
include "pan/functions";

"/" = create("clusterdata/openstack_devel",
  "metadata", value("/metadata")
);
include "archetype/base";
include "personality/openstack_hypervisor/config";
include "archetype/final";
```

In a host profile, you would have:

```
include "cluster/openstack_devel/client";
```
which mainly does the personality include.

So the issue remains valid for me...

**Comment by ned21** [comment]
Have you tried setting the archetype to non-compilable? I think that achieves the same thing.

**Comment by jouvin** [comment]
@ned21 yes, you are right but if you do this you cannot use 'aq make cluster' anymore to rebuild all the hosts in the cluster...

**Comment by ned21** [comment]
Interesting if that is a feature of the broker, I always assumed it was a side-effect of the PAN compilation because there are dependencies from the cluster object to the host objects.

`aq reconfigure --membersof $cluster` is the command you want to rebuild all the hosts in a cluster.

**Comment by jouvin** [comment]
I missed that, thanks! Then marking the cluster archetype does the job. I'll update my [doc PR](https://github.com/quattor/quattor.github.com/pull/259) about clusters then!
